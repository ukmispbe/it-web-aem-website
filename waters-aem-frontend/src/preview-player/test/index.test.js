import React from "react";
import { shallow } from "enzyme";
import PreviewPlayer from "../index";

describe("<PreviewPlayer />", () => {
  test("should render correctly", () => {
        const defaultProps = {    
            imgSrc: '',
            widths: [],
            defaultImage: '',
            zoomIcon: '/',
            zoomIconText: 'Click or Tap to Zoom',
            openModal: () => {},         
            defaultImage: '', 
            videoConfig: {}   
    };
    const wrapper = shallow(<PreviewPlayer {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should render image with label", () => {
    const props = {
        imgSrc: 'http://localhost:4502/content/dam/waters/en/brand-assets/product/alliance-launch/alliance-online.jpg.{{width}}.resize/img.jpg',
        widths: ['1280','770','620','375','320','256','140','128'],
        defaultImage: 'http://localhost:4502/content/dam/waters/en/brand-assets/product/alliance-launch/alliance-online.jpg.770.resize/img.jpg',
        zoomIcon: '/',
        zoomIconText: 'Click or Tap to Zoom',
        openModal: () => {},         
        defaultImage: '', 
        videoConfig: {}
    };
    const wrapper = shallow(<PreviewPlayer {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  
});