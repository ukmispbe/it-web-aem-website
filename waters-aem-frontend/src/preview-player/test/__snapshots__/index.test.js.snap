// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PreviewPlayer /> should render correctly 1`] = `ShallowWrapper {}`;

exports[`<PreviewPlayer /> should render image with label 1`] = `ShallowWrapper {}`;
